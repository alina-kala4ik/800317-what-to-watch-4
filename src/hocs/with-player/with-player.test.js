import React from "react";
import renderer from "react-test-renderer";
import withPlayer from "./with-player.jsx";
import PropTypes from "prop-types";

const film = {
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  screenshotSrc: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  posterSrc: `img/bg-the-grand-budapest-hotel.jpg`,
  movieCoverSrc: `img/the-grand-budapest-hotel-poster.jpg`,
  genre: `Drama`,
  yearRelease: 2017,
  description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.`,
  rating: 5.6,
  numberVotes: 278,
  producer: `Wes Andreson`,
  actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  runTime: 99,
  videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
};

const MockComponent = (props) => {
  const {children} = props;

  return <div>
    {children}
  </div>;
};

MockComponent.propTypes = {
  children: PropTypes.node.isRequired,
};

const MockComponentWrapped = withPlayer(MockComponent);

it(`render withPlayer`, ()=>{
  const tree = renderer.create(
      <MockComponentWrapped film={film}>
        <video />
      </MockComponentWrapped>, {
        createNodeMock: () => {
          return {};
        }}
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
