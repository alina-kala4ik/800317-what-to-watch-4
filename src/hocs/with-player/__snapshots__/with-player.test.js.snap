// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render withPlayer 1`] = `
<div>
  <video
    autoPlay={true}
    className="player__video"
    loop={false}
    poster="img/fantastic-beasts-the-crimes-of-grindelwald.jpg"
    src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  />
</div>
`;
